de.dlr.sc.virsat.project.ui/src/de/dlr/sc/virsat/project/ui/navigator/dropAssist/DVLMObjectDropAdapterAssistant.java

```diff
@@ -9,98 +9,176 @@
+		defaultInheritanceDropAdapter = new DVLMDefaultInheritanceDropAdapterAssistant();
+		defaultCopyPasteDropAdapter.init(getContentService());
+	}
+
 	/**
-	 * Method to create a Drop Command based on the Copy Paste functionality
-	 * @param target The object to witch to drop to
-	 * @param operation the type of DND operation
-	 * @return the ccommand if it is creatable or null in case it could not be created
+	 * Reads out the extension point to initialize the map of
+	 * concept ids vs. drop adapters
 	 */
-	private Command createDropCommand(Object target, int operation) {
-		Command dropCommand = null;
-		
-		if (target instanceof EObject) {
-			EObject eObject = (EObject) target;
-			VirSatTransactionalEditingDomain virSatEd = VirSatEditingDomainRegistry.INSTANCE.getEd(eObject);
-			ISelection selection = LocalSelectionTransfer.getTransfer().getSelection();
-			
-			if (selection instanceof IStructuredSelection) {
-				IStructuredSelection structerdSelection = (IStructuredSelection) selection;
-				
-				@SuppressWarnings("unchecked")
-				Collection<Object> dndObjects = structerdSelection.toList();
-					
-				if (operation == DND.DROP_COPY) {
-					dropCommand = VirSatPasteFromClipboardCommand.create(virSatEd, eObject, dndObjects, ClipboardState.COPY);
-				} else if (operation == DND.DROP_MOVE) {
-					dropCommand = VirSatPasteFromClipboardCommand.create(virSatEd, eObject, dndObjects, ClipboardState.CUT);
+	protected void initDropAssistantMap() {
+		// Re initialize the map of drop assistants
+		mapConceptIdToDelegateDropAdapter = new HashMap<>();
+	
+		// Now start looping over all Navigator extensions to find the possible registered drop assistants
+		IConfigurationElement[] navigatorExtensions = Platform.getExtensionRegistry().getConfigurationElementsFor(EXTENSION_POINT_NAVIGATOR);
+		for (IConfigurationElement navigatorExtension : navigatorExtensions) {
+			if (EXTENSION_POINT_DROP_ASSISTANT_CONTRIBUTION.equals(navigatorExtension.getName())) {
+				String conceptId = navigatorExtension.getAttribute(EXTENSION_POINT_DROP_ASSISTANT_CONCEPT_ID);
+				CommonDropAdapterAssistant dropAdapter;
+				try {
+					dropAdapter = (CommonDropAdapterAssistant) navigatorExtension.createExecutableExtension(EXTENSION_POINT_DROP_ASSISTANT_IMPLEMENTATION);
+					dropAdapter.init(getContentService());
+					if (!mapConceptIdToDelegateDropAdapter.containsKey(conceptId)) {
+						mapConceptIdToDelegateDropAdapter.put(conceptId, dropAdapter);
+					} else {
+						Activator.getDefault().getLog().log(new Status(Status.ERROR, Activator.getPluginId(), "Cannot register two drop adapter for the same concept: " + conceptId));
+					}
+				} catch (CoreException e) {
+					Activator.getDefault().getLog().log(new Status(Status.ERROR, Activator.getPluginId(), "Failed to create executable extension for drop adapter in concept: " + navigatorExtension.getContributor() + "->" + conceptId + " because: " + e.getMessage()));

```

Comment:
I think we should also pass the stack trace to the logger.

Reply:
Yes added it to the places where INFOs are now logged as WARNINGS